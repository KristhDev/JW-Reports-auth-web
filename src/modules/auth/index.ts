export * from './components';
export * from './hooks';
export * from './layouts';
export * from './pages';
export * from './routes';