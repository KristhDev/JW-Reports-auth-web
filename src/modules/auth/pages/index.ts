export { Confirm } from './Confirm';
export { ResetPassword } from './ResetPassword';