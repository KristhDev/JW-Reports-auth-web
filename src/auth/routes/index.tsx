import { RouteObject } from 'react-router-dom';

import { Confirm, ResetPassword } from '../pages';

export const routes: RouteObject[] = [
    {
        path: '',
        element: <Confirm />
    },
    {
        path: 'reset-password',
        element: <ResetPassword />
    }
];