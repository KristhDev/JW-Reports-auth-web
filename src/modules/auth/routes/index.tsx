import { RouteObject } from 'react-router-dom';

/* Pages */
import { Confirm, ResetPassword } from '../pages';

/* Creating a route object. */
export const routes: RouteObject[] = [
    {
        path: '',
        element: <Confirm />
    },
    {
        path: 'reset-password',
        element: <ResetPassword />
    }
];