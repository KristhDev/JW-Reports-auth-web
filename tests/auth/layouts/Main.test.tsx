import { render } from '@testing-library/react';

import { Main } from '../../../src/auth/layouts';

describe('Test in <Main /> Layout', () => {
    it('should to match snapshot', () => {
        const { container } = render(
            <Main title="Title Test" />
        );

        expect(container).toMatchSnapshot();
    });

    it('should render respective props', () => {
        const { getByRole } = render(
            <Main
                title="Title Test"
            >
                <h3>SubTitle test</h3>
            </Main>
        );

        const titleHeader = getByRole('heading', { level: 1 });
        const subTitleHeader = getByRole('heading', { level: 3 });

        expect(titleHeader).toBeTruthy();
        expect(titleHeader.innerHTML).toBe('Title Test');
        expect(subTitleHeader).toBeTruthy();
        expect(subTitleHeader.innerHTML).toBe('SubTitle test');
    });
});