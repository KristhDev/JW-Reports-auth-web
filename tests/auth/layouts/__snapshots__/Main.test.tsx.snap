// Vitest Snapshot v1

exports[`Test in <Main /> Layout > should to match snapshot 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="title-container undefined"
      data-testid="title-container"
    >
      <h1
        class="title-text undefined"
      >
        Title Test
      </h1>
    </div>
  </div>
</div>
`;
